Update documentation to build Wazuh dashboard package (#7961)
wazuh/wazuh-documentation: source/development/packaging/generate-dashboard-package.rst

      # git clone -b <BRANCH_OR_TAG> https://github.com/wazuh/wazuh-dashboard.git
      # cd wazuh-dashboard/
      # yarn osd bootstrap
      # yarn build < --linux || --linux-arm > --skip-os-packages --release

   Example:

   .. code:: console

      # git clone -b 4.10.2 https://github.com/wazuh/wazuh-dashboard.git
      # cd wazuh-dashboard/
      # yarn osd bootstrap
      # yarn build --linux --skip-os-packages --release

   .. note::

>       If you want to build a package to ARM architecture, you must use the ``--linux-arm`` flag in the ``yarn build`` command instead of the ``--linux`` flag.

Reviewer: ```suggestion
      To build a package for ARM architecture, use the ``--linux-arm`` flag with the ``yarn build`` command, rather than the ``--linux`` flag.
```

Run the ``build-packages.sh`` script in the ``dev-tools/build-packages/`` folder of the repository. The script requires the following parameters:

> - ``-a``, ``--app``: Path to the ``wazuh-package.zip``.
> - ``-b``, ``--base``: Path to the ``dashboard-package.zip``.
> - ``-s``, ``--security``: Path to the ``security-package.zip``.
> - ``-v``, ``--version``: Set the version of this build.
> - ``--all-platforms``: Build all platforms.
> - ``--deb``: Build deb.
> - ``--rpm``: Build rpm.
> - ``--tar``: Build tar.gz.
> - ``-r``, ``--revision``: [Optional] Set the revision of this build. By default, it is set to 1.
> - ``--production``:[Optional] The naming of the package will be ready for production. Otherwise, it will include the hash of the current commit.
> - ``--arm``: [Optional] Build for arm64 instead of x64.
> - ``--debug``: [Optional] Enables debug mode, which will show detailed information during the script run.
> - ``--silent``: [Optional] Enables silent mode, which will show the minimum possible information during the script run. ``--debug`` has priority over this.

Reviewer: ```suggestion
- ``-a``, ``--app``: Indicate the path to the ``wazuh-package.zip``.
- ``-b``, ``--base``: Indicate the path to the ``dashboard-package.zip``.
- ``-s``, ``--security``: Indicate the path to the ``security-package.zip``.
- ``-v``, ``--version``: Set the version for this build.
- ``--all-platforms``: Build the package for all platforms.
- ``--deb``: Build a deb package.
- ``--rpm``: Build an rpm package.
- ``--tar``: Build a tar.gz archive.
- ``-r``, ``--revision``: [Optional] Indicate the revision for this build; defaults to 1.
- ``--production``:[Optional] Prepare the package naming for production; otherwise, it will include the hash of the current commit.
- ``--arm``: [Optional] Build for arm64 architecture instead of x64.
- ``--debug``: [Optional] Enables debug mode to display detailed information during script execution.
- ``--silent``: [Optional] Enables silent mode to show minimal information during the script run; note that --debug takes precedence over this option.
```

.. code:: console

   # cd ../wazuh-dashboard/dev-tools/build-packages/
   # ./build-packages.sh -v <VERSION> -r <REVISION> --<DISTRIBUTION_(--deb || --rpm || --tar || --all-platforms)> -a file:///<PATH_TO_wazuh-package.zip> -s file:///<PATH_TO_security-package.zip> -b file:///<PATH_TO_dashboard-package.zip>

.. note::

>    In the inputs where a local path is available, use ``file://<absolute_path>`` to indicate it.

Reviewer: ```suggestion
   In cases where a local path is available, use ``file://<absolute_path>`` to indicate it.
```

Author: Done https://github.com/wazuh/wazuh-documentation/pull/7961/commits/4bf50dd4c253ec17a96007a99a3fcb53b7091fba

.. note::

>    To build ``arm`` packages, you need to run the script in an arm machine, and use an arm build of the wazuh-dashboard base with ``-b``

Reviewer: ```suggestion
   To build ``arm`` packages, you need to run the script on an arm machine and use an arm build of the wazuh-dashboard base with the ``-b`` option.```

Author: Done https://github.com/wazuh/wazuh-documentation/pull/7961/commits/4bf50dd4c253ec17a96007a99a3fcb53b7091fba

Example:

.. code:: console

   # cd ../wazuh-dashboard/dev-tools/build-packages/
   # ./build-packages.sh -v 4.10.2 -r 1 --deb -a file:///packages/wazuh-package.zip -s file:///packages/security-package.zip -b file:///packages/dashboard-package.zip

The package will be generated in the ``output`` folder of the same directory where the script is located.

Build with Docker image
^^^^^^^^^^^^^^^^^^^^^^^

> This option facilitates the creation of packages of all plugins in order to generate the Wazuh dashboard installer.

Reviewer: ```suggestion
This option facilitates the creation of packages that include all necessary plugins to generate the Wazuh dashboard installer.
```

Author: Done https://github.com/wazuh/wazuh-documentation/pull/7961/commits/4bf50dd4c253ec17a96007a99a3fcb53b7091fba

.. topic:: Requirements

>    - A system with Docker.
>    - Internet connection (to download the docker images the first time).

Reviewer: ```suggestion
   - A system with Docker.
   - Internet connection (to download the Docker images the first time).
```

Author: Done https://github.com/wazuh/wazuh-documentation/pull/7961/commits/4bf50dd4c253ec17a96007a99a3fcb53b7091fba

.. topic:: Steps

>    #. Clone the Wazuh dashboard repository and move to the ``dev-tools/build-packages/base-packages-to-base`` folder of the repository.

Reviewer: ```suggestion
   #. Clone the Wazuh dashboard repository and navigate to the ``dev-tools/build-packages/base-packages-to-base`` folder within it.
```

Author: Done https://github.com/wazuh/wazuh-documentation/pull/7961/commits/4bf50dd4c253ec17a96007a99a3fcb53b7091fba

      .. code:: console
      
         # git clone -b <BRANCH_OR_TAG> https://github.com/wazuh/wazuh-dashboard.git
         # cd wazuh-dashboard/dev-tools/build-packages/
      
      Example:
      
      .. code:: console
      
         # git clone -b 4.10.2 https://github.com/wazuh/wazuh-dashboard.git
         # cd wazuh-dashboard/dev-tools/build-packages/base-packages-to-base

   #. Run the script ``run-docker-compose.sh`` with the following parameters:

>       -  ``--node-version``: Node version to use in the ``.nvmrc`` file.
>       -  ``-b``, ``--base``: Branch of the Wazuh dashboards repository.
>       -  ``-a``, ``--app``: Branch of the Wazuh dashboards Plugins repository.
>       -  ``-s``, ``--security```: Branch of the Wazuh Security Dashboards Plugin repository.
>       -  ``--arm``: [Optional] Build for arm64 instead of x64.

Reviewer: ```suggestion
      -  ``--node-version``: Specifies the Node version for the ``.nvmrc`` file.
      -  ``-b``, ``--base``: Branch of the Wazuh dashboards repository.
      -  ``-a``, ``--app``: Branch of the Wazuh dashboards Plugins repository.
      -  ``-s``, ``--security```: Branch of the Wazuh Security Dashboards Plugin repository.
      -  ``--arm``: [Optional] Build for arm64 architecture instead of x64.
```

Author: Done https://github.com/wazuh/wazuh-documentation/pull/7961/commits/4bf50dd4c253ec17a96007a99a3fcb53b7091fba

      .. code:: console
      
         # docker build \
         # --node-version <NODE_VERSION> \
         # --base <BRANCH_OF_wazuh-dashboard> \
         # --app <BRANCH_OF_wazuh-dashboard-plugins> \
         # --security <BRANCH_OF_wazuh-security-dashboards-plugin>

      Example:
      
      .. code:: console
      
         # bash run-docker-compose.sh \
         #   --app 4.10.2 \
         #   --base 4.10.2 \
         #   --security 4.10.2 \
         #   --node-version 18.19.0

>    #. The packages will be generated in the ``packages`` folder of the ``base-packages-to-base`` folder.

Reviewer: ```suggestion
   #. The packages will be created in the ``packages`` directory within the ``base-packages-to-base`` folder.
```

Author: Done https://github.com/wazuh/wazuh-documentation/pull/7961/commits/4bf50dd4c253ec17a96007a99a3fcb53b7091fba

      .. note::

>          If you want to build a custom package, you can replace the packages in the ``packages`` folder with the ones you have customized.

Reviewer: ```suggestion
         To build a custom package, replace the contents of the ``packages`` folder with your customized packages.
```

Author: Done https://github.com/wazuh/wazuh-documentation/pull/7961/commits/4bf50dd4c253ec17a96007a99a3fcb53b7091fba

   #. Zip the packages

      .. code:: console

         # cd ./packages
         # zip -r -j ./dashboard-package.zip ./wazuh-dashboard/*.tar.gz
         # zip -r -j ./security-package.zip ./wazuh-security-dashboards-plugin/*.zip
         # zip -r -j ./wazuh-package.zip ./wazuh-dashboard-plugins/*.zip

   #. Build deb, rpm, or tar.gz packages

      .. code:: console

         # cd ../../
         # ./build-packages.sh -v <VERSION> -r <REVISION> (optional --arm) --<DISTRIBUTION_(--deb || --rpm || --tar || --all-platforms)> -a file:///<PATH_TO_wazuh-package.zip> -s file:///<PATH_TO_security-package.zip> -b file:///<PATH_TO_dashboard-package.zip>

      Example:

      .. code:: console

         # ./build-packages.sh -v 4.10.2 -r 1 --deb -a file://$(pwd)/base-packages-to-base/packages/wazuh-package.zip -s file://$(pwd)/base-packages-to-base/packages/security-package.zip -b file://$(pwd)/base-packages-to-base/packages/dashboard-package.zip

>       The package will be generated in the ``output`` folder of the same directory where the script is located.

Reviewer: ```suggestion
      The package will be created in the ``output`` folder within the same directory as the script.
```

Author: Done https://github.com/wazuh/wazuh-documentation/pull/7961/commits/4bf50dd4c253ec17a96007a99a3fcb53b7091fba
